### ISSUE TITLE
Error: Is a directory (os error 21), readfile 'public/' with deno template

### ISSUE THREAD

**Issue by vinaybedre** [remix-run/remix#5625]
### What version of Remix are you using?

1.14.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Run `npx create-remix@latest --template=deno --typescript`

Open the project, run `npm install` and run `npm run dev`

```
deno --version

deno 1.31.1 (release, aarch64-apple-darwin)
v8 11.0.226.13
typescript 4.9.4
```

### Expected Behavior

The remix project to start and working on http://localhost:8000

### Actual Behavior

```
Error: Is a directory (os error 21), readfile 'public/'
    at async Object.readFile (internal:runtime/js/40_read_file.js:23:18)
```

**Comment by sstiglitz** [comment]
I ran into this as well.

The issue only occurs on the root route. If you go to another route, it will serve successfully.

[Here's the relevant code](https://github.com/remix-run/remix/blob/main/packages/remix-deno/server.ts#L79-L81). The way the catch is setup, it is looking for specific codes, however when I run a simple example, there isn't a code property on the error object:

```ts
Deno.readFile('./public').catch((error) => {
  console.log(error.code)
})

// Result code undefined
```

I think it needs to be switched to look at the message. Something like:

```
if (error.message.startsWith('Is a directory') || error.message.startsWith('No such file or directory')) {
  throw new FileNotFoundError(filePath);
}
```

**Comment by sstiglitz** [comment]
Hey there, I feel it can be merged in at any time. I see it's been approved, might just be waiting on another release round or something.

**Comment by machour** [comment]
I just tried on 1.15.0 and I'm not seeing any error, http://localhost:8000/ loads fine for me

```
deno --version
deno 1.32.5 (release, aarch64-apple-darwin)
v8 11.2.214.9
typescript 5.0.3
```

I wonder if the root issue got fixed in deno 1.32.0: https://github.com/denoland/deno/pull/17958

@vinaybedre could you try with an up to date deno?

**Comment by vinaybedre** [comment]
The issue is fixed. Hence closing
